pytorch_lightning/utilities/auto_restart.py

```diff
@@ -80,12 +88,13 @@ def __iter__(self) -> Iterator[Any]:
                 break
 
         self._current_iteration = 0
+        self._cached_state_dict = None
         self.restarting = False
 
     def __len__(self) -> int:
         return len(self._sampler)
 
-    def state_dict(self, num_batches_processed: Optional[int] = None) -> Dict[int, Dict[str, int]]:
+    def state_dict(self, num_batches_processed: Optional[int] = None) -> Dict[int, Dict[str, Any]]:
```

Comment:
Why change the type hint from int?

Reply:
artifact from a larger PR where this will contain more than int in the future.

Reply:
```suggestion
    def state_dict(self, num_batches_processed: Optional[int] = None) -> Dict[int, Dict[str, int]]:
```